dvc/remote/base.py

```diff
@@ -685,17 +685,21 @@ def _checkout_file(
         self, path_info, checksum, force, progress_callback=None
     ):
         cache_info = self.checksum_to_path_info(checksum)
-        if self.exists(path_info):
-            msg = "data '{}' exists. Removing before checkout."
-            logger.warning(msg.format(str(path_info)))
-            self.safe_remove(path_info, force=force)
 
-        self.link(cache_info, path_info)
-        self.state.save_link(path_info)
-        self.state.save(path_info, checksum)
+        if self._needs_checkout(path_info, checksum):
+            if self.exists(path_info):
+                msg = "data '{}' exists. Removing before checkout."
+                logger.warning(msg.format(str(path_info)))
+                self.safe_remove(path_info, force=force)
+            self.link(cache_info, path_info)
+            self.state.save_link(path_info)
+            self.state.save(path_info, checksum)
```

Comment:
Can you readd blank line you removed? It looks harder to parse (with an eye) now.

Reply:
done

Reply:
@pared did you forget to `push`?